cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -358,7 +358,10 @@ public final ProgramController run(final Program program, ProgramOptions oldOpti
           .setClassLoader(MainClassLoader.class.getName());
 
         // Add namespace details
-        twillPreparer.withConfiguration(getNamespaceConfigs(program.getNamespaceId()));
```

Comment:
Don't you need to call twillPreparer.withConfiguration() in the non-tethered case (RUNTIME_NAMESPACE arg is not set)?

Reply:
good point. Added this back in for the non-tethered case